src/main/java/com/iexec/sms/secret/teetaskruntime/AppRuntimeSecretController.java

```diff
@@ -0,0 +1,315 @@
+        }
+
+        final Ownable appContract = iexecHubService.getOwnableContract(appAddress);
+        if (appContract == null || !Objects.equals(appContract.getContractAddress(), appAddress)) {
+            log.debug("App does not exist" +
+                            " [requesterAddress:{}, appAddress:{}]",
+                    requesterAddress, appAddress);
+            return ResponseEntity
+                    .status(HttpStatus.NOT_FOUND)
+                    .body(createErrorPayload("App does not exist"));
+        }
+
+        if (teeTaskRuntimeSecretService.isSecretPresent(
+                OnChainObjectType.APPLICATION,
+                appAddress,
+                SecretOwnerRole.REQUESTER,
+                requesterAddress,
+                secretIndex)) {
+            log.debug("Can't add requester secret as it already exists" +
+                            " [requesterAddress:{}, appAddress:{}, secretIndex:{}]",
+                    requesterAddress, appAddress, secretIndex);
+            return ResponseEntity
+                    .status(HttpStatus.CONFLICT)
+                    .body(createErrorPayload("Secret already exists"));
+        }
+
+        final Optional<TeeTaskRuntimeSecretCount> oAllowedSecretsCount =
+                teeTaskRuntimeSecretCountService.getMaxAppRuntimeSecretCount(
+                        appAddress,
+                        SecretOwnerRole.REQUESTER
+                );
+
+        if (oAllowedSecretsCount.isEmpty()) {
+            log.error("Can't add requester secret as no secret count has been provided" +
+                            " [requesterAddress:{}, appAddress:{}, secretIndex:{}]",
+                    requesterAddress, appAddress, secretIndex);
+            return ResponseEntity
+                    .badRequest()
+                    .body(createErrorPayload("No secret count has been provided"));
+        }
+
+        final Integer allowedSecretsCount = oAllowedSecretsCount.get().getSecretCount();
+        if (secretIndex >= allowedSecretsCount) {
+            log.error("Can't add requester secret as index is greater than allowed secrets count" +
+                            " [requesterAddress:{}, appAddress:{}, secretIndex:{}, secretCount:{}]",
+                    requesterAddress, appAddress, secretIndex, allowedSecretsCount);
+            return ResponseEntity
+                    .badRequest()
+                    .body(createErrorPayload("Index is greater than allowed secrets count"));
+        }
+
+        teeTaskRuntimeSecretService.encryptAndSaveSecret(
+                OnChainObjectType.APPLICATION,
+                appAddress,
+                SecretOwnerRole.REQUESTER,
+                requesterAddress,
+                secretIndex,
+                secretValue
+        );
+        return ResponseEntity.noContent().build();
+    }
+
+    @RequestMapping(method = RequestMethod.HEAD, path = "/requesters/{requesterAddress}/apps/{appAddress}/secrets/{secretIndex}")
```

Comment:
I think we need an endpoint that returns the value of count, in case the developer forgets it or if another service needs to do some verification. It can be merged with this endpoint:
* `count == 0` not set
* `count > 0` set + actual value

Reply:
I don't get how you would merge your suggestion with this endpoint. It only checks whether a secret exists or not.
BTW, if we add this behavior, returning `0` as a not-set value seems hazardous. An app developer could explicitly say the app won't manage any requester secret - that would ensure there is no injection that would lead to a security issue.

Reply:
We can use a separate endpoints, of course, if you prefer.
But one endpoint seems enough, IMHO. If the developer calls ` GET /..../secret-count`, they'll receive a value:
* if the value is `0` than, there is no secrets (at the end there is no difference if the user has explicitly set it or not)
* if the value is `> 0` then the count has been configured and we know it's value.

Reply:
I still don't understand which endpoint you're referring to :sweat_smile: 
There's no `GET /..../secret-count` endpoint as of now. We only provide those 3 endpoints:
- `POST /apps/{appAddress}/requesters/secrets-count`
- `HEAD /apps/{appAddress}/secrets/{secretIndex}`
- `HEAD /requesters/{requesterAddress}/apps/{appAddress}/secrets/{secretIndex}`

The first one lets the app developer set the max secrets count for its app, while the two lasts simply tell whether a secret has been set for given index. I don't see any of them being able to return an integer value.
However, adding a new `GET /..../secret-count` endpoint would perfectly work. I'm on it!

Reply:
Endpoint added in https://github.com/iExecBlockchainComputing/iexec-sms/pull/69/commits/965151414a9117d8267afa86fea403509ea85c55

Reply:
Your last commit summarizes what I wanted to say. Sorry if it wasn't that clear and thanks.